lib/lintSource.js

```diff
@@ -117,6 +130,10 @@ function lintPostcssResult(stylelint, postcssResult, config) {
 	const postcssDoc = postcssResult.root;
 
 	if (postcssDoc) {
+		if (!('type' in postcssDoc)) {
+			throw new Error('Unexpected Postcss root object!');
+		}
+
```

Comment:
An real `Root` object should have `type: 'root'`.

https://github.com/postcss/postcss/blob/7.0.26/lib/postcss.d.ts#L1042

Reply:
Maybe we should remove the exclamation mark from here too @jeddy3 ?